pandas-stubs/core/indexes/accessors.pyi

```diff
@@ -192,7 +314,65 @@ class PeriodProperties(DatetimeAndPeriodProperties):
         how: Literal["E", "END", "FINISH", "S", "START", "BEGIN"] = ...,
     ) -> PeriodArray: ...
 
+class PeriodProperties(
+    Properties,
+    _PeriodProperties,
+    _DatetimeFieldOps[Series[int]],
+    _IsLeapYearProperty,
+    _FreqProperty[BaseOffset],
+): ...
+
 class CombinedDatetimelikeProperties(
-    DatetimeProperties, TimedeltaProperties, PeriodProperties
+    DatetimeProperties[
+        Series[int],
+        Series[bool],
+        Series,
+        Series[dt.date],
+        Series[dt.time],
+        str,
+        TimestampSeries,
+        Series[str],
+        PeriodSeries,
+    ],
+    _TimedeltaPropertiesNoRounding,
+    _PeriodProperties,
 ):
     def __new__(cls, data: Series): ...
+
+class TimestampProperties(
```

Comment:
Could mark newly introduced classes that are not in the implementation, maybe `_TimestampProperties`? Same for `TimestampProperties` and `PeriodSeries`

Reply:
> Could mark newly introduced classes that are not in the implementation, maybe `_TimestampProperties`? Same for `TimestampProperties` and `PeriodSeries`

Not sure.  Since `TimestampProperties` is the result of `TimestampSeries.dt`, we are then importing it there.  Do we want to have imports of "private" classes internally within the stubs?

So what I chose to do here is make any class that gets imported by another stub file public (no underscore) for the purpose of the stubs.

Reply:
okay, sounds reasonable.